ouroboros-network/src/Ouroboros/Network/PeerSelection/Churn.hs

```diff
@@ -77,79 +114,59 @@ peerChurnGovernor tracer deadlineChurnInterval bulkChurnInterval psOverallTimeou
         writeTVar churnModeVar mode
         return mode
 
-    -- TODO: #3396 revisit the policy for genesis
-    increaseActivePeers :: ChurnMode -> STM m ()
-    increaseActivePeers mode =
+    increaseActivePeers :: ChurnRegime -> PeerSelectionTargets -> STM m ()
+    increaseActivePeers mode base =
         modifyTVar peerSelectionVar (\targets -> targets {
           targetNumberOfActivePeers =
-              case mode of
-                   ChurnModeNormal  ->
-                       targetNumberOfActivePeers base
-                   ChurnModeBulkSync ->
-                       min 2 (targetNumberOfActivePeers base)
-        })
+              if mode == Default
+              then targetNumberOfActivePeers base
+              else min 2 (targetNumberOfActivePeers base) })
```

Comment:
I don't think this is right.  In bootstrap & legacy mode we have to take into account `ChurnMode`.  This allso applies to other diff hunks in this file.

if we want to use a single type instead of `ChurnMode` & `UseBoostrapPeers` & `UseGenesis` flag, maybe we should define `ChurnRegime` as:

```
data ChurnRegime = ChurnGenesis
                 | ChurnBootstrap ChurnMode
                 | ChurnLegacy ChurnMode
```

Reply:
Is this not handled properly by the pickChurnRegime function, which takes into account ChurnMode and UseBootstrapPeers? So in legacy mode, or Praos, when ChurnModeBulkSync is on, regardless of value of UseBootstrapPeers, the second branch is taken, just as before.

Reply:
I missed that, that's nice.

Reply:
I don't like using `Ord` instance for `ChurnRegime` though. Either document why the `Ord` instance is essential or just use a pattern matching (I think the latter is better).

Note using `a > X` (or something similar) is as bad to use as `_` in:
```
catch a of
  X -> ...
  _ -> ...
```
If we add a new constructor we won't get a warning / error.  It's a little bit more verbose to pattern match on all constructors, but it avoids silly mistakes that can easily be missed in reviews.